src/Microsoft.Extensions.Logging.AzureWebAppDiagnostics/Internal/AzureBlobSink.cs

```diff
@@ -70,6 +70,13 @@ public class AzureBlobSink : PeriodicBatchingSink
         /// <inheritdoc />
         protected override async Task EmitBatchAsync(IEnumerable<LogEvent> events)
         {
+            await EmitBatchInternalAsync(events);
+        }
+
+
+        // For unit testing
+        internal async Task EmitBatchInternalAsync(IEnumerable<LogEvent> events)
```

Comment:
you could make EmitBatchAsync `protected internal`. Potentially you could avoid InternalsVisibleTo if you derived from AzureBlobSink in your tests and created a public EmitBatchAsyncPublich method that forwards to EmitBatchAsync...